presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java

```diff
@@ -764,10 +767,25 @@ public void testQueryLoggingCount()
         });
     }
 
+    private <T> T waitUtilStable(Supplier<T> computation, Duration timeout)
+    {
+        T lastValue = computation.get();
+        long start = System.nanoTime();
+        while (!currentThread().isInterrupted() && nanosSince(start).compareTo(timeout) < 0) {
+            sleepUninterruptibly(100, MILLISECONDS);
+            T currentValue = computation.get();
+            if (currentValue.equals(lastValue)) {
+                return currentValue;
+            }
+            lastValue = currentValue;
+        }
+        throw new RuntimeException("timeout");
```

Comment:
nit: Should we throw a TimeoutException here?

Reply:
I used `UncheckedTimeoutException` from Guava.